### DISCUSSION TITLE
BOM Pricing pie charts for Variants

### DISCUSSION THREAD

**Discussion by Geekly** [inventree/InvenTree#4156] [Q&A] [answered]
I'm trying to view pricing for variant BOMs. I'm referring specifically to the pie chart that shows the contribution of each part in a BOM to the total price. The total under Pricing Information seems correct, it's just the pie chart that's incomplete.

Template Part BOM
- Part A (Inherited)
- Part B

The BOM Pricing pie chart for the Template Part is complete and correct. Both parts show up in the pie chart with prices.

Variant 1 (based on Template Part)
- Part A (Inherited)
- Part C

The BOM Pricing pie chart for Variant 1 is incomplete. Only the Part C shows up in the pie chart and non of the inherited parts. Obviously we want to see all of the parts. 

Does anyone have suggestions about what I could be doing wrong? Is this the right place for this post?

Thanks very much.

**Comment by matmair** [comment, marked as answer]
@Geekly this sounds like ti could be a bug. Would you mind opening a bug report? Please also add (if you can) screenshots of the BOM tables of the parts (the first few columns should be enough).

**Reply by Geekly** [reply]
Will do. I was able to reproduce situation in a simple test case. I noticed that pricing information did not become available until I marked the part as "Saleable". I did those for both Parts that were assemblies.

**Reply by matmair** [reply]
@Geekly I think is intended behaviour but @SchrodingersGat is more experienced in that section as he recently worked on it.

**Reply by Geekly** [reply]
That doesn't surprise me, it just hung me up for a bit when I was trying to add the part prices. Maybe they need to be Purchasable OR Saleable for the pricing to appear. That makes sense.
